core/templates/dev/head/directives.js

```diff
@@ -164,3 +164,49 @@ oppia.directive('mobileFriendlyTooltip', ['$timeout', function($timeout) {
     }
   };
 }]);
+
+oppia.directive('handleDropdownNavigation', function () {
```

Comment:
Can we get rid of this directive altogether, and use something like ng-keypress instead in the TopNavigationBarDirective controller? We are trying to get rid of attribute-type directives in general.

Reply:
yes, we can get rid of the directive, but the only problem with using ng-keydown is we have to add it in 6 places(3 places for each drop-down).
1.parent of the submenu to handle Enter keypress.
2.First "li" of the submenu to handle Shift + Tab.
3.Last "li" of the submenu to handle when the user clicks Tab on the last list item.
We can use `ng-keydown="handleNavigation($event,'aboutMenu','enter','open')"`
Please see this [gist](https://gist.github.com/srikar0896/cd93eca667878a2e3948bed49c8824c3), which will handle the keydown event.

Reply:
The general idea of the gist looks good to me!

Reply:
Done.Removed the directive and added a function to handle navigation in the directive controller.